csrc/host_ir/jit.cpp

```diff
@@ -602,6 +661,125 @@ class HostIrCompileDispatcher : public OptInDispatch {
     val_to_value_[out_tv] = out_tensor;
   }
 
+  void handle(MatmulOp* matmul_op) final {
+    llvm::Module* module = builder_.GetInsertBlock()->getParent()->getParent();
+
+    llvm::Value* t_a =
+        getOrCreateValue(matmul_op->inA(), val_to_value_, builder_);
+    llvm::Value* t_b =
+        getOrCreateValue(matmul_op->inB(), val_to_value_, builder_);
+    llvm::Value* t_out =
+        getOrCreateValue(matmul_op->out(), val_to_value_, builder_);
+
+    builder_.CreateCall(
+        module->getFunction(kMatmulOutFuncName), {t_out, t_a, t_b});
+    val_to_value_[matmul_op->out()] = t_out;
+  }
+
+  void handle(LinearOp* linear_op) final {
+    llvm::Module* module = builder_.GetInsertBlock()->getParent()->getParent();
+    llvm::LLVMContext& context = builder_.getContext();
+
+    llvm::Value* t_in =
+        getOrCreateValue(linear_op->inA(), val_to_value_, builder_);
+    llvm::Value* t_weight =
+        getOrCreateValue(linear_op->inB(), val_to_value_, builder_);
+    llvm::Value* t_out =
+        getOrCreateValue(linear_op->out(), val_to_value_, builder_);
+
+    llvm::Value* t_bias = nullptr;
+    if (linear_op->hasBias()) {
+      t_bias = getOrCreateValue(linear_op->bias(), val_to_value_, builder_);
+    } else {
+      // Create a proper null pointer for LLVM
+      auto* tensor_type = getTensorPtrType(context);
+      t_bias = llvm::ConstantPointerNull::get(
+          llvm::cast<llvm::PointerType>(tensor_type));
+    }
+    builder_.CreateCall(
+        module->getFunction(kLinearOutFuncName),
+        {t_out, t_in, t_weight, t_bias});
+    val_to_value_[linear_op->out()] = t_out;
+  }
+
+  // Launch Kernel Function LLVM IR Generation
+  void handle(hir::LaunchKernel* launch_kernel) final {
+    llvm::Module* module = builder_.GetInsertBlock()->getParent()->getParent();
+    llvm::LLVMContext& context = builder_.getContext();
+    auto* void_ptr_type = getInt8PtrType(context);
+    auto* void_array_ptr_type = getInt8PtrDynamicArrayType(context);
+
+    // Convert input TensorViews to void pointers and get tensor pointers
+    llvm::SmallVector<llvm::Value*, 1> input_tensors;
+    for (auto* tv : launch_kernel->inputs()) {
+      input_tensors.push_back(getOrCreateValue(tv, val_to_value_, builder_));
+    }
+
+    // Convert output TensorViews to void pointers and get tensor pointers
+    llvm::SmallVector<llvm::Value*, 1> output_tensors;
+    for (auto* tv : launch_kernel->outputs()) {
+      output_tensors.push_back(getOrCreateValue(tv, val_to_value_, builder_));
+    }
+
+    // Get the cacheId from the main function's first argument
+    llvm::Value* cache_id_arg =
+        getOrCreateValue(launch_kernel->cacheId(), val_to_value_, builder_);
```

Comment:
:clap: 

Let's put a comment here that this is to align with implementation/protocol established in HostIrEvaluator